Maple2.Server.Game/PacketHandlers/UserEnvHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Maple2.PacketLib.Tools;
+using Maple2.Server.Core.Constants;
+using Maple2.Server.Core.PacketHandlers;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+
+namespace Maple2.Server.Game.PacketHandlers;
+
+public class UserEnvHandler : PacketHandler<GameSession> {
+    public override RecvOp OpCode => RecvOp.RequestUserEnv;
+
+    private enum Command : byte {
+        ChangeTitle = 1,
+    }
+
+    public override void Handle(GameSession session, IByteReader packet) {
+        var command = packet.Read<Command>();
+        switch (command) {
+            case Command.ChangeTitle:
+                HandleChangeTitle(session, packet);
+                break;
+        }
+    }
+
+    private void HandleChangeTitle(GameSession session, IByteReader packet) {
+        int titleId = packet.ReadInt();
+
+        if (!session.Player.Value.Unlock.Titles.Contains(titleId) && titleId != 0) {
```

Comment:
nit: check `titleId != 0` first to short-circuit this check since that's the cheaper & more common case.